### DISCUSSION TITLE
Suggestion: move the username to the top right side

### DISCUSSION THREAD

**Discussion by Verhoeckx** [cockpit-project/cockpit#16154] [Ideas]
The last week there have been a few occasions that I wanted to now under which username I was logged in (as a normal user or as root). When I start searching for it, I automatically started looking at the top right of the screen because that's where the dummy image of the avatar is visible.

My suggestion:
Move the username from the top left side of the screen (username@hostname) to the top right side of the screen (at the right side of the avatar). You can keep the hostname at the top left side.

**Comment by marusak** [comment]
It used to be the case - I found this old screenshot: https://cockpit-project.org/images/service-details.png

The issue is, that when you switch to a different machine, then the user you logged in as, can differ from who you are on the current (switched) system. That causes inconsistency. Also users who use ssh are used to `user@machine` so having machine on left and user at right, without any obvious connection seems like step back.

**Reply by Verhoeckx** [reply]
Okay, all good valid points!
Maybe it's an idea to replace the dummy avatar with a hamburger menu (or with an icon of three vertical dots)?

**Reply by marusak** [reply]
Right, now the icon represents "Session" (You can pick language, add ssh keys or log out). So different icon would actually make sense. Not sure if hamburger is the right option as that is used often for collapsing the main menu, nor kebab as that seems a little too inexpressive. @garrett WDYT?

**Reply by marusak** [reply]
Actually, maybe hamburger would work? For example firefox uses hamburger on top right to toggle settings menu.

**Reply by Verhoeckx** [reply]
I also doubt if a hamburger menu is the right icon but at the moment I don't know of something better. I agree that the 'kebab menu' :smile:  isn't expressive enough. 

I thought the avatar was a placeholder for a real photo and that the function to upload an image wasn't added yet :flushed: .

**Reply by Verhoeckx** [reply]
The problem with a hamburger menu is that it maybe promises to much: a complete main menu. This is why I think a 'kebab menu' could be used. But maybe there is a different icon that is applicable?

**Reply by garrett** [reply]
We used to have a "hamburger menu" in Cockpit, but only in mobile mode. We had since restyled to the empty avatar, just as PF4 does.

https://www.patternfly.org/v4/components/page/design-guidelines#masthead

> 4. User menu: The user menu is always the right-most item in the masthead. It includes an avatar and the username of the currently logged in user. The dropdown items may vary based on user settings and permissions available. Common items to include are: User name, Account ID or Account management. This section should always include a Log out option at the bottom of the dropdown list. If there is a need to allow users to change application display language, it should appear in this dropdown menu as well.

However, we do not use the hamburger icon to show/hide nav like they suggest, as it doesn't make sense in Cockpit. And if we did use a hamburger menu icon in Cockpit, then we would be reusing an icon that looks like other PatternFly apps that does a different thing, so we should avoid doing so.

And kebab icons are used in-page, not in the navbar really. They're usually meant to be used for "other stuff" like "other actions" (and in some cases the only actions you can choose aside from interacting with a table row, for example).

So what could we use? We could use a gear icon in its place, perhaps...? 

> Settings: The settings menu contains ways to configure and customize an application. Interacting with settings either takes users to a settings area or reveals a dropdown menu with more options. The settings menu is represented by the fa-cog icon.

They are session settings, after all. It's not really system-account things. (Previously, Cockpit did have a quick link to the logged in user's account management page in the menu too, as well as help.) 

Language is a session-related setting. SSH keys modifies the system, but is still user related and vital to the session—however, it is a setting too. Log out is definitely session-related, but _not_ a setting, though.

So a gear isn't really a 100% fit either, but it might be the closest familiar concept aside from a user picture (which we have now).

---

I'd like to point out most sites have a pattern of a user picture in the top-right. 

Google does this across their websites and many of their apps. 

![Screenshot 2021-07-29 at 12-51-18 Google](https://user-images.githubusercontent.com/10246/127479505-4f716d6b-421b-489e-ac57-ec51becc6980.png)
![Screenshot 2021-07-29 at 12-50-45 Google Docs](https://user-images.githubusercontent.com/10246/127479509-ce040f74-490b-4c35-8dbb-a49b99535823.png)
![Screenshot 2021-07-29 at 12-50-18 cockpit project - YouTube](https://user-images.githubusercontent.com/10246/127479512-b8f0c76b-2cc7-4ba0-89c0-a82ec3ee0a4e.png)


So does Flickr.

![Screenshot 2021-07-29 at 12-57-58 Home Flickr](https://user-images.githubusercontent.com/10246/127480458-864ab8f7-5a75-4d96-97bc-e966fe607e08.png)



Facebook does this as well, with notifications to the right of the user related settings.

![Screenshot 2021-07-29 at 13-29-08 (20+) Facebook](https://user-images.githubusercontent.com/10246/127484496-cb78b54e-6686-4479-a298-e7a68a7a2e1b.png)



Amazon has a user menu with settings in the top-right too (although with a shopping cart in the far right). 

![Screenshot 2021-07-29 at 12-54-40 Amazon de Low Prices in Electronics, Books, Sports Equipment more-2](https://user-images.githubusercontent.com/10246/127480325-62ab3203-ee66-4861-8598-ab7e5f5b0796.png)


Twitter used to do this too, but they moved the same concept to the bottom-left... except in mobile, where it's the top-left. But it's exclusively account settings, like switching accounts. Their settings are under the "More" menu.

My point is that it's a _very common_ pattern across most of the Web. So it might not be a big problem overall. The big issue is that it's a generic profile image with no way to change it and that it's mainly tied to a session, not an account, so having an account icon is a bit misleading. 

Some of the sites use a name, some use a picture, some use both... but we're using a blank picture and no name (now). Although the name is on the left, but that's an account on a system.

---

It's probably worth nothing Firefox uses a _gear_ in the top-right of the start page to change the settings of what you see there.

![Screenshot from 2021-07-29 12-46-57](https://user-images.githubusercontent.com/10246/127479069-83729019-0cde-4101-832c-765381d8550b.png)

![Screenshot from 2021-07-29 12-47-17](https://user-images.githubusercontent.com/10246/127479077-37a56c3e-4a7a-48df-a215-fdc1a886b922.png)

Chrome and Edge use an account icon in their toolbar with a menu (which is similar to Firefox's menu that you mention). But Edge also has in-page settings on the start page in the shape of a gear, just like Firefox. Brave also has a gear for editing, but it's in the bottom-right.

![Screenshot from 2021-07-29 13-06-42](https://user-images.githubusercontent.com/10246/127484326-95990dca-01bb-4524-849a-9c22e35bfdd7.png)


---

Additionally, I don't think people had issues with this in the usability studies we ran, but it's been a while and I don't think we specifically tested for this. Plus, memories change over time, so I could also be misremembering it too.

**Reply by garrett** [reply]
We could change the top-right to look like this:

![cockpit-overview-session (copy)](https://user-images.githubusercontent.com/10246/127490160-128dfc7f-d42e-4c05-965b-9e65d5137eb5.png)

I noticed that PF now suggests a filled circle for the heading, wheras the line version is for in-page help icons. This is new. We should update. I've updated it in this mockup. With the outline version that we're currently using, the cog looks out of place, but it fits right in with the solid version.

![Screenshot 2021-07-29 at 14-18-13 PatternFly 4 • Icons](https://user-images.githubusercontent.com/10246/127490468-678cf71b-8777-4428-950a-7ee16aef3251.png)


In context of the rest of the page, it would look like this:

![cockpit-overview-session](https://user-images.githubusercontent.com/10246/127490252-d1ca833c-45c3-46c2-b9e7-e36bff7a141a.png)

---

One problem with the cog: We're using it in the metrics settings. This would mean two cogs right above each other. They'd each have different words, but it'd probably still be a little odd.

Here's a screenshot:

![Screenshot 2021-07-29 at 14-19-39 Overview - garrett Bolt](https://user-images.githubusercontent.com/10246/127490713-6f05b396-fc1b-4156-8a77-f06f8ea47414.png)

And here's a modification to make it show what it would look lik eeven with the stacking:

![Screenshot 2021-07-29 at 14-19-39 Overview - garrett Bolt-mod](https://user-images.githubusercontent.com/10246/127491041-e3765a0b-2f70-41e5-8e49-b54cbca7bd80.png)

Additionally, someone might think the cog is a flower. But that's fine. Whatever, it's the session and settings flower then. :white_flower: :grin: 

If you have a suggestion for something else, PatternFly's icon library is browsable at: https://www.patternfly.org/v4/guidelines/icons/#all-icons

**Reply by Verhoeckx** [reply]
I think the idea to rename it to `Session` and cog wheel icon to it, is a really good idea! Look much more understandable now!

Another option would be to use the `fa-clock` icon but the problem with this icon is that a user now probably thinks he/she can find the time in the menu. So in the end this is not a good option.

I think having two cogs near each other is not really a problem: I think the user will understand the difference. In doubt you could use the `fa-wrench` icon (for the metric settings)?

I went through the list of all icons but couldn't find a better, more appropriate one.

Random thought: I just realized that the logo of Cockpit isn't used anywhere in the application. Not a problem but maybe it could be added somewhere? So the user knows he/she is using Cockpit.

**Reply by garrett** [reply]
> Another option would be to use the fa-clock icon but the problem with this icon is that a user now probably thinks he/she can find the time in the menu. So in the end this is not a good option.

Yeah, I thought of that too, with the same issue as well. It's not really time related.

> I think having two cogs near each other is not really a problem: I think the user will understand the difference. In doubt you could use the fa-wrench icon (for the metric settings)?

I was also wondering about this. It's probably not really a big deal, especially as both have text right after the icon.

> I went through the list of all icons but couldn't find a better, more appropriate one.

Same.

> Random thought: I just realized that the logo of Cockpit isn't used anywhere in the application. Not a problem but maybe it could be added somewhere? So the user knows he/she is using Cockpit.

We purposefully don't mention the name Cockpit anywhere but the about dialog. We should improve the about dialog though; there's an open issue on that: #15961.

---

By-the-way, I converted this dicussion into an issue @ #16173.

**Comment by Verhoeckx** [comment]
> We purposefully don't mention the name Cockpit anywhere but the about dialog. We should improve the about dialog though; there's an open issue on that: #15961

What is the reason for this? Now the application looks a little bit anonymous.
New design looks nice but it also doesn't fit completely in the current Cockpit design.

> By-the-way, I converted this dicussion into an issue @ #16173.

Great! I just noticed that the help menu, strangely enough, now also becomes more visible.

**Comment by Verhoeckx** [comment]
Maybe a stupid question but what is the difference between the section `Authorized public SSH keys` in the `Accounts` overview and the option `SSH keys` in the session menu? What is the latter used for?

**Reply by martinpitt** [reply]
They are for "opposite ends" of SSH: The accounts page's user details shows the users ~/.ssh/authorized_keys, i.e. public keys that can ssh into that user's account on this machine. The session menu's SSH keys shows the SSH public/private key pairs of the user who is logged into cockpit, i.e. the keys which you can use to ssh *to* some other machines.

**Reply by Verhoeckx** [reply]
Ah, thanks! I now understand what it is used for!
Maybe rename it? To something like `Server keys`?
It's difficult to find a good name for it.

And are they session related (it's in the session menu) or are they permanent?

**Reply by martinpitt** [reply]
"Server" keys is misleading IMHO -- these are keypairs for the client side of SSH. (A "server key" sounds more like an SSH host key.) The dialog explains this some more: "Use the following keys to authenticate against other systems".

"SSH identity keys" perhaps, but that's getting rather long for the menu.
